contributingGuides/CONTRIBUTING.md

```diff
@@ -111,11 +111,11 @@ Additionally if you want to discuss an idea with the open source community witho
 3. If you cannot reproduce the problem, pause on this step and add a comment to the issue explaining where you are stuck or that you don't think the issue can be reproduced.
 
 #### Propose a solution for the job
-4. After you reproduce the issue, make a proposal for your solution and post it as a comment in the corresponding GitHub issue (linked in the Upwork job). Your solution proposal should include a brief written technical explanation of the changes you will make. Include "Proposal" as the first word in your comment. 
+4. After you reproduce the issue, complete the [proposal template here](./PROPOSAL_TEMPLATE.md) and post it as a comment in the corresponding GitHub issue (linked in the Upwork job).
     - Note: Before submitting a proposal on an issue, be sure to read any other existing proposals. Any new proposal should be substantively different from existing proposals.
-5. Pause at this step until someone from the Contributor-Plus team and / or someone from Expensify provides feedback on your proposal (do not create a pull request yet).
-6. If your solution proposal is accepted by the Expensify engineer assigned to the issue, Expensify will hire you on Upwork and assign the GitHub issue to you.
-7. Once hired, post a comment in the Github issue stating when you expect to have your PR ready for review 
+5. Refrain from leaving additional comments until someone from the Contributor-Plus team and / or someone from Expensify provides feedback on your proposal (do not create a pull request yet).
+6. If your proposal is accepted by the Expensify engineer assigned to the issue, Expensify will hire you on Upwork and assign the GitHub issue to you.
+7. Once hired, post a comment in the Github issue stating when you expect to have your PR ready for review.
```

Comment:
Getting-your-thoughts: Should we say something about only posting one proposal at a time? Oftentimes I see someone will come along and post up to as many as four different proposals. On high-traffic issues, it's a lot to manage.

Reply:
I think that's a good call out. I think ideally a contributor should only have one  "proposal waiting for a review" at a time. Sometimes it happens that you come up with alternate solutions. But I'd argue those belong back the proposal you have open and you should update the current proposal if it hasn't been reviewed yet.

Which also makes me wonder whether we want people to "update their original proposal" or post a new one (since people don't all do this the same way).

Thinking more on all of this and I think my ideal would probably be something like:

- Always wait for a response to your current proposal before leaving additional proposals.
- It's OK to update a proposal if it hasn't been reviewed yet.
- If you're making a new proposal or updating a proposal then edit your original proposal and leave a comment like:

```
## Proposal

[Updated](link to proposal)
```

Reply:
>Always wait for a response to your current proposal before leaving additional proposals.

maybe that is unclear though. because we are saying it's ok to leave additional proposals. I think it would be better to just have one proposal that the contributor evolving until it's perfect.

Reply:
Maybe something like...

- Do not leave more than one proposal.
- Do not make extensive changes to your current proposal until after it has been reviewed.
- If you want to make an entirely new proposal or update an existing proposal please go back and edit your original proposal and leave a comment like this:

```
## Proposal
[Updated](link to proposal)
```

Reply:
I like the idea of always keeping 1 post per proposal and only having one proposal reviewed at a time. I don't quite get the `link to proposal` when updating an existing one. If it's all in the same post, then there isn't anything to link to. Plus, GH keeps the edit history of the post anyway.

The thing I dislike is when there is one large proposal that keeps getting tweaked, and so the tweaked proposal exists in several different forms across comments and it's difficult to see what changed between them.

That's why I like just always keeping it in a single post and making updates to the existing post rather than posting a new comment.

Reply:
>If it's all in the same post, then there isn't anything to link to

Ah, so in my head it was a link to the comment which "jumps" you to the proposal in the same issue.

I think the idea with posting an update to a proposal is just that it encourages the best practice of letting everyone know your proposal is "ready for review" again. And then we'd have a general loop of:

- Propose
- Incorporate feedback
- Update proposal
- Request re-review
- Etc.

This probably happens organically already (perhaps inconsistently) so maybe the attempt to create a process for this seems pointless.

Reply:
ah, I got you. Just like how we post "updated" on a PR to let people know it can be reviewed again. I like that